### PR TITLE
Add e2e test to verify knative examples

### PR DESCRIPTION
**Pull request by savitaashture**
# Changes
Fixes https://github.com/tektoncd/triggers/issues/1089
PR adds changes to install Knative Serving and runs `custom-resource` examples for both `v1alpha1` and `v1beta1`

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Triggers now verify Knative based tests
```

### CODE REVIEW

**Comment by savitaashture** [comment]
Hi @dibyom could you take a look at this PR please

File: test/e2e-test-knative-examples.sh [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  # port forwarding to access application
+  kubectl -n kourier-system port-forward service/kourier $forwardingPort:$forwardingPort > /dev/null 2>&1 &
+  sleep 1
+  hostURL=$(kubectl get el ${elName} -o=jsonpath='{.status.address.url}')
+  host=$(echo $(echo $hostURL | tr "://" "\n") | cut -d' ' -f 2)
+
+  curl -v \
+   -H 'X-GitHub-Event: pull_request' \
+   -H 'X-Hub-Signature: sha1=ba0cdc263b3492a74b601d240c27efe81c4720cb' \
+   -H 'Content-Type: application/json' \
```
Comment (dibyom) [inline test/e2e-test-knative-examples.sh:86]: Can we put this in a curl.sh file like we do for other examples.

**Review by dibyom — COMMENTED** [review]
Thanks @savitaashture Looks good overall. My main comment is can we this logic into e2e-test-examples.sh itself instead of creating a new file and copying stuff over? i.e. we can add in a couple of new functions in that file itself (`curl_knative_service`, `install_knative_serving`) and then add a new function called `run_knative_tests` that is basically the `main` from this file?

**Comment by savitaashture** [comment]
@dibyom I have addressed comments PTAL

**Pull request merged**